### ISSUE TITLE
Hidden album contents still shown within the "Uncategorized" section

### ISSUE THREAD

**Issue by ialexpw** [ente-io/photos-app#1647]
**Describe the bug**
I've recently moved a lot of my media over to ente and it's been pretty painless so far, although one strange thing is happening and I'm unsure if it's me or not!

I've sorted contents into multiple albums and then I've set a few to "Hidden" so they don't show up on the main screen. In the app's main screen this seems to work OK, although if I go to Albums I have two entries for "Uncategorized" (see screenshot), one has all of my images (including those within Hidden albums that I can just view) and one has (the correct) images that are currently unsorted.

**To Reproduce**
1. Import content
2. Add albums and move content around
3. Set some albums to Hidden
4. Check the two Uncategorized sections

**Expected behavior**
There is no way to access the Hidden albums unless you authenticate into the Hidden section and the Uncategorized sections should match with the amount of images

**Screenshots**
![Screenshot_20240112_093957_ente Photos](https://github.com/ente-io/photos-app/assets/7994724/a0c348ff-d802-423d-ae24-9d4ebee1493e)

**Smartphone (please complete the following information):**
Samsung S23 Ultra, latest Android UI 6

**Additional context**
This may be related, but on ente web the main image feed shows ALL of my images also (even from the Hidden folders, so I presume this is loading from the full "Uncategorized" section, while ente app is showing the correct main feed without the Hidden items. I've tried the clean option, but it is still the same.

**Comment by ialexpw** [comment]
@vishnukvmd Further to your email I'm attempting to export the entire library then I may try and clean & re-import later (unless you need me to check anything else!)

When exporting it does show two "Uncategorized" folders, even though neither were created by myself.

![folders](https://github.com/ente-io/photos-app/assets/7994724/0569deea-457f-44e2-b520-cca9032e83b4)

**Comment by ialexpw** [comment]
Unsure if related, but I removed & re-added all of my media. It created the second "Uncategorized" album again, so I moved out everything from the album that was created so there's only one and removed the empty one.

Now once I have some of my media in hidden folders again (done from the mobile app), I can see all of those images that should be hidden on the main wall on both the web app and the desktop app, but on the mobile app it's showing as it should.

**Comment by vishnukvmd** [comment]
Hey @ialexpw, sorry for the delay, we've been trying to reproduce this issue, without much luck.

Could you please let us know which platform you are using to perform the initial import?

We'd like to isolate the cause for duplicate Uncategorized album, and once we do, it will help us understand why hidden items are misbehaving.

Thank you!

**Comment by ialexpw** [comment]
The initial upload was from the Desktop application (Windows 11), I added a folder with ~16,000 images in and it asked for the initial folder so I selected Uncategorized (the only option as it was a fresh account).

After the upload was complete, it had created a "Uncategorized" album and put some of the images inside (not all of them).

I'll create a test account with a smaller sub-set of images and see if I can recreate the issue easily. I'll update once I try it this evening.

**Comment by ialexpw** [comment]
To update;

I didn't manage to recreate the extra album creation, but I can recreate the hidden files/albums still showing.

1. I created a new account and uploaded 3 images on the web app.
2. I hid an image directly from the mobile app (no album)
3. This disappeared from both the web and mobile app (all OK)
4. I created an album on the mobile app & moved an image into it
5. I then chose to hide this album from the mobile app
6. The image within this album is still shown on the main gallery on the web app, but not the mobile app

I've shown some images below to hopefully show it. It seems to hide the album, but not the items inside.

![aLnO3B](https://github.com/ente-io/photos-app/assets/7994724/773f53cc-4279-4f14-bd26-4491ce72fba9)

![Screenshot_20240119_115020_ente Photos](https://github.com/ente-io/photos-app/assets/7994724/827a530d-ac2c-40c2-8806-723de3dcdfe7)

![Screenshot_20240119_115127_ente Photos](https://github.com/ente-io/photos-app/assets/7994724/aa8404a0-08c5-4733-9a66-f74f2bc7989b)
